magma/lib/magma/model.rb

```diff
@@ -1,9 +1,7 @@
 class Magma
   class Model < Sequel::Model
     class << self
-      Magma::Attribute.descendants.
-        reject { |attribute| attribute == Magma::ForeignKeyAttribute }.
-        each do |attribute|
+      Magma::Attribute.type_attributes.each do |attribute|
```

Comment:
I'm curious; what is this change?

Reply:
I noticed there is no method collecting the actual list of magma types; because of a weird inheritance pattern all the descendants of Magma::Attribute are type classes except ForeignKeyAttribute, which is a base class for some others. The only place this logic was codified was here, so i just isolated it in a method. At some point maybe ForeignKey could be refactored into a module for consistency.